app/assets/javascripts/discourse/app/controllers/composer.js

```diff
@@ -1481,13 +1488,22 @@ export default Controller.extend({
 
   toggleFullscreen() {
     this._saveDraft();
-    if (this.get("model.composeState") === Composer.FULLSCREEN) {
-      this.set("model.composeState", Composer.OPEN);
+
+    const composer = this.model;
+
+    if (composer?.viewFullscreen) {
+      composer?.set("composeState", Composer.OPEN);
     } else {
-      this.set("model.composeState", Composer.FULLSCREEN);
+      composer?.set("composeState", Composer.FULLSCREEN);
+      composer?.toggleProperty("showFullScreenExitPrompt");
     }
   },
 
+  @discourseComputed("model.viewFullscreen", "model.showFullScreenExitPrompt")
+  fullScreenPrompt(isFullscreen, showExitPrompt) {
+    return isFullscreen && showExitPrompt;
```

Comment:
Can we make sure (here or elsewhere) that this isn't shown on mobile? Probably better to not show on all touch devices IMO.

Reply:
`viewFullscreen` is only available in the desktop view because we don't add the full-screen toggle on mobile. That toggle is the only way to enter that state beside the full screen composer keyboard shortcut (<kbd>Shift</kbd> + <kbd>F11</kbd>).

Great catch RE touch devices, I added a check for that :+1: